### PR TITLE
fix: rules of hooks lint warnings

### PR DESCRIPTION
**Pull request by bryce-mcmath**
# Summary of Changes

This is the second wave of fixes for react lint warnings, this time fixing `react-hooks/rules-of-hooks` warnings and then making them errors going forward. I will do a final wave with `react-hooks/exhaustive-deps` but there was beginning to be too many changes at once.

In refactoring some tests I found a good approach for testing hooks, please take a look at the `Proof bundle resolver works correctly` test if you need inspo for testing hooks in the future.

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/core/App/components/tour/TourOverlay.tsx [hunk 1]
```diff
@@ -116,7 +116,7 @@ export const TourOverlay = (props: TourOverlayProps) => {
         testID={testIdWithKey('SpotTooltip')}
         style={{ ...tooltipStyle, opacity: 1, position: 'absolute' }}
       >
-        <tourStep.render
+        <tourStep.Render
```
Comment (jleach) [inline packages/legacy/core/App/components/tour/TourOverlay.tsx:119]: Curious, what's the diff if `render` works?

File: packages/legacy/core/App/utils/helpers.ts [hunk 2]
```diff
@@ -267,20 +258,21 @@ export function getConnectionName(
...
   if (credential.connectionId) {
-    const connection = useConnectionById(credential.connectionId)
     return connection?.alias || connection?.theirLabel || credential.connectionId
   }
 
   return 'Unknown Contact'
 }
 
-export function getConnectionImageUrl(connectionId: string) {
+export function useConnectionImageUrl(connectionId: string) {
```
Comment (jleach) [inline packages/legacy/core/App/utils/helpers.ts:275]: Checking, this is a hook ya? AFAIK only hooks should start with `use`?

**Review by jleach — APPROVED** [review]

**Comment by jleach** [comment]
LGTM. Only curious about renaming the fn from `get*` to `use*` and visa-versa. Curious if only hooks should start with `use` and that's why the rename?

File: packages/legacy/core/App/components/tour/TourOverlay.tsx (hunk 1 above)
Comment (bryce-mcmath) [reply packages/legacy/core/App/components/tour/TourOverlay.tsx:119]: The linter complains that React components should start with a capital letter

File: packages/legacy/core/App/utils/helpers.ts (hunk 2 above)
Comment (bryce-mcmath) [reply packages/legacy/core/App/utils/helpers.ts:275]: It is a hook, as it uses a hook, so it had to be named with `use`

**Comment by bryce-mcmath** [comment]
> LGTM. Only curious about renaming the fn from `get*` to `use*` and visa-versa. Curious if only hooks should start with `use` and that's why the rename?

@jleach yes, that is exactly the reason for the name changes. Some functions were hooks but didn't start with `use` and some functions weren't hooks but did start with `use`

**Pull request merged**